src/AzureExtension/DataManager/AzureDataManagerCache.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Dynamic;
+using DevHomeAzureExtension.Client;
+using DevHomeAzureExtension.DataManager;
+using DevHomeAzureExtension.DataModel;
+using DevHomeAzureExtension.DeveloperId;
+using Microsoft.TeamFoundation.Core.WebApi;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+using Microsoft.VisualStudio.Services.Account;
+using Microsoft.VisualStudio.Services.Account.Client;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Serilog;
+
+namespace DevHomeAzureExtension;
+
+public partial class AzureDataManager
+{
+    private static readonly Uri VSAccountUri = new(@"https://app.vssps.visualstudio.com/");
+
+    private static readonly int PullRequestProjectLimit = 50;
+
+    private static readonly int PullRequestRepositoryLimit = 25;
+
+    public async Task UpdateDataForAccountsAsync(RequestOptions? options = null, Guid? requestor = null)
+    {
+        // A parameterless call will always update the data, effectively a 'force update'.
+        await UpdateDataForAccountsAsync(TimeSpan.MinValue, options, requestor);
+    }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+    public async Task UpdateDataForAccountsAsync(TimeSpan olderThan, RequestOptions? options = null, Guid? requestor = null)
```

Comment:
Why is this marked async?

Reply:
It had been changed at times where it was and was not using async calls, and may yet have them in the future, and I got tired of constantly changing the signature, the method, and the interface to being async/non-async when there was no functional difference between them, so I suppressed the warning and left it there.

Reply:
I totally sympathize with it going back and forth while you were working on it, but now that it will theoretically be stable for a bit, I'd suggest correcting to non-async.

Reply:
I made an attempt to change it and the references to non async, and as I did that I encountered the real reason I marked this async - it chains through several other methods (which also need to change signature) and is ultimately called by the Update method in the dataupdater class, and bifurcating that code into async and non async just for this one method, along with a chain of other classes and methods seems like a lot of unnecessary code and changes, especially if an async method call ends up needing to be added later at some point and it all becomes dead code. And since it will just run synchronously and is functionally equivalent and this is a warning, not an error, for the sake of simplicity of the code I want to leave it as-is.